gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -102,9 +101,7 @@ public SeekableByteChannel position(long newPosition) throws IOException {
         return this;
       }
       position = newPosition;
-      size = fetchSize(gcsService, file);
-      channel.close();
-      channel = gcsService.openReadChannel(file, position);
+      channel.seek((int) newPosition);
```

Comment:
If the unit tests didn't catch that error earlier, would it be possible for us to add a test that does?